### PR TITLE
Bug fix json editor flat

### PR DESCRIPTION
**Pull request by nelsonGuti**
## Summary

Before:

https://user-images.githubusercontent.com/62297014/178605572-5a6d5f7d-6675-44db-b03f-998531f18282.mov

After:

https://user-images.githubusercontent.com/62297014/178605826-239d843d-21c8-4417-955d-28b5a3f67683.mov

## Checklist

- [ ] \*Added unit tests
- [x] \*Added a code reviewer
- [x] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [ ] Updated the demo page
- [x] Included screenshots of visual changes

_\*required_

### CODE REVIEW

**Review by jogaj — APPROVED** [review]

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts [hunk 1]
```diff
@@ -82,16 +82,27 @@ export class ObjectNodeFlatComponent extends ObjectNode implements OnInit, OnCha
   }
 
   onUpdatePropertyName(options: { id: string; name: string }): void {
-    const existingSchemaProperty = this.schemaRef.properties[options.name];
+    let schema = this.schemaBuilderMode ? this.schemaRef : this.schema;
+
+    if (!schema?.properties) {
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts:87]: This is fine... but I prefer:

```
schema ||= {};
schema.properties ||= {};
```

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts [hunk 2]
```diff
@@ -82,16 +82,27 @@ export class ObjectNodeFlatComponent extends ObjectNode implements OnInit, OnCha
...
     const existingPropertyValue = this.model[options.name];
     const oldName = this.propertyIndex[options.id].propertyName;
 
     this.duplicatedFields.delete(options.id);
 
     if (!existingSchemaProperty && existingPropertyValue === undefined) {
-      const index = Object.keys(this.schemaRef.properties).findIndex(prop => prop === oldName);
-      this.updateSchemaPropertyName(this.schemaRef, options.name, this.propertyIndex[options.id].propertyName);
-      this.swapSchemaProperties(index);
+      const index = Object.keys(schema.properties).findIndex(prop => prop === oldName);
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts:101]: Is this the same as `Object.keys(schema.properties).indexOf(oldName)`?

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts [hunk 3]
```diff
@@ -247,6 +264,7 @@ export class ObjectNodeFlatComponent extends ObjectNode implements OnInit, OnCha
 
   private updateSchemaPropertyName(schema: JSONEditorSchema, newName: string, oldName: string): void {
     this.updateRequiredProperties(schema, newName, oldName);
+    schema.properties = (schema || {}).properties || {};
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts:269]: Seams to me if `schema` is undefiend (hence `schema || {}`) then `schema.properties =` will fail.  I'd suggest the same as suggested above.

**Review by Hypercubed — APPROVED** [review]
LGTM.. a couple of minor suggestions.

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts (hunk 2 above)
Comment (nelsonGuti) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts]: it is yep, changed it.

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts (hunk 3 above)
Comment (nelsonGuti) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts]: Added the null check here as well. Also added it on the `updateRequiredProperties()` which had a similar check. Thanks!

**Review by Hypercubed — APPROVED** [review]

**Pull request merged**